wallet-new/src/Cardano/Wallet/Kernel/Restore.hs

```diff
@@ -239,50 +335,48 @@ restoreWalletHistoryAsync wallet rootId target tgtSlot prefilter = do
 
         -- Skip EBBs
         whenRight block $ \mb -> do
-          -- Filter the blocks by account
-          blund <- (Right mb, ) <$> getUndoOrThrow genesisHash hh
-          (prefilteredBlocks, txMetas) <- prefilter blund
-
-          -- Apply the block
-          k    <- getSecurityParameter (wallet ^. walletNode)
-          ctxt <- withNode $ mainBlockContext genesisHash mb
-          mErr <- update (wallet ^. wallets) $
-                    ApplyHistoricalBlock k ctxt prefilteredBlocks
-          case mErr of
-            Left err -> throwM $ RestorationApplyHistoricalBlockFailed err
-            Right () -> return ()
-
-          -- Update our progress
-          slotCount <- getSlotCount (wallet ^. walletNode)
-          let flat             = flattenSlotId slotCount
-              blockPerSec      = MeasuredIn . BlockCount . perSecond <$> rate
-              throughputUpdate = maybe identity (set wriThroughput) blockPerSec
-              slotId           = mb ^. mainBlockSlot
-          updateRestorationInfo wallet wId ( (wriCurrentSlot .~ flat slotId)
-                                           . (wriTargetSlot  .~ flat tgtSlot)
-                                           . throughputUpdate )
-
-          -- Store the TxMetas
-          forM_ txMetas (putTxMeta (wallet ^. walletMeta))
-
-        -- Get the next block from the node and recurse.
-        if target == hh then
-          finish
-        else
-          nextBlock hh >>= \case
-            Nothing      -> throwM $ RestorationFinishUnreachable target hh
-            Just header' -> restore genesisHash header' timing'
+            -- Filter the blocks by account
+            blund <- (Right mb, ) <$> getUndoOrThrow genesisHash hh
+            (prefilteredBlocks, txMetas) <- prefilter blund
+
+            -- Apply the block
+            k    <- getSecurityParameter (wallet ^. walletNode)
+            ctxt <- withNode $ mainBlockContext genesisHash mb
+            mErr <- update (wallet ^. wallets) $
+                   ApplyHistoricalBlock k ctxt prefilteredBlocks
+            case mErr of
+                Left err -> throwM $ RestorationApplyHistoricalBlockFailed err
```

Comment:
Do we catch this error somewhere? Where do we monitor the restoration workers?

Reply:
It's possible that this is okay, this can only happen if the _node_ switches to a fork (I think), i which case we will restart this anyway. But needs some thought and some comments.